packages/contracts/deploy/new/10_framework/42_plugin_setup_processor_upgradeable.ts

```diff
@@ -0,0 +1,43 @@
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {getContractAddress, skipIfNotZkSync} from '../../helpers';
+
+import pluginSetupProcessorUpgradeableFactoryArtifact from '../../../artifacts/src/zksync/PluginSetupProcessorUpgradeable.sol/PluginSetupProcessorUpgradeable.json';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const {deployments, ethers} = hre;
+  const {deploy} = deployments;
+  const [deployer] = await ethers.getSigners();
+
+  // Get `PluginRepoRegistry` address.
+  const pluginRepoRegistryAddress = await getContractAddress(
+    'PluginRepoRegistry',
+    hre
+  );
+
+  // Get `Management DAO` address.
+  const managementDAOAddress = await getContractAddress('DAO', hre);
+
+  await deploy('PluginSetupProcessor', {
```

Comment:
```suggestion
  await deploy('PluginSetupProcessorUpgradeable', {
```
is there a reason why you didn't put the `Upgradeable` name?

Reply:
> better not to name them as differently, otherwise, in every place we decide to use its deployment, we would need to figure out first which network it is and fetch the desired one accordingly which I don't like.

I think we really should be explicit here. The earlier we catch a potentially old artifact flying around, the better. 
I suppose it wouldn't work, but let's make sure that no accidents can happen here.
Please rename

Reply:
Well, There're 6 places we use the following:

```
await getContractAddress('PluginSetupProcessor', hre);
```

If I do what you're suggesting, then in all 6 places, I have to do:

```
const pspContractName = ZK_SYNC_NETWORKS.includes(hre.network.name) 
        ? 'PluginSetupProcessorUpgradeable' 
        : 'PluginSetupProcessor'

  let pluginSetupProcessorAddress = await getContractAddress(
    pspContractName,
    hre
  );
```

I really hate this approach 👍

Reply:
Wrap in a function then, seems like a perfect use case

Reply:
https://github.com/aragon/osx/pull/596/commits/9cd35fc54b54e8cc8a4e89b8d5770435c80b6ffa